SofaKernel/modules/SofaBaseTopology/GridTopology.cpp

```diff
@@ -223,16 +238,34 @@ void GridTopology::setSize(int nx, int ny, int nz)
 void GridTopology::checkGridResolution()
 {
     const Vec3i& _n = d_n.getValue();
-    if (_n[0] < 2 || _n[1] < 2 || _n[2] < 2)
+
+    int dim = 0;
+    bool badDim = false; // In case one of the dimension is null.
+
+    for (int i=0; i<3; i++)
+    {
+        if (_n[i] < 1){
+            badDim = true;
+            break;
+        }
+        else if(_n[i] > 2)
+            dim++;
+    }
+
+    m_gridDim = (Grid_dimension)dim;
+
+    if (badDim)
     {
         msg_warning(this) << "The grid resolution: ["<< _n[0] << " ; " << _n[1] << " ; " << _n[2] <<
```

Comment:
If the object inherit from Base you can now write 
msg_warning() instead of msg_warning(this)